edx/edx-app-ios
# [feature] add video download qualities options

### Description

[LEARNER-8503](https://openedx.atlassian.net/browse/LEARNER-8503)

Give learners the choice to download whichever video quality they want to download.

### How to test this PR
- [x] UI: Settings Page

-  Add the video download quality selection option on the Settings screen

- 4 options to select the download quality are available when the user taps into the video download quality selection option

- mobile_low_360p, mobile_high _540p, desktop_mp4 (720p):  all the mappings

- [ ]  UI: Videos Page

- A new section will be added to the video downloads page where the user can see the currently selected download quality option

- Tapping on this section will deeplink to the quality selector (1.b.)

- The download size will be calculated based on the selected download quality option

 - [ ] Defaults & Persistence 

- The default selection will be ‘Auto’

- Current ‘Auto' behavior is mobile_low → mobile_high → fallback downloads firsts

- Keep this behavior the same for the “auto” selection

- Once the user selects an option, it will persist until the user selects another option. 

- The selected option will be applicable to all courses for that user on that device.

- If the selected quality is not available for a video, the next lower quality will be used to download. If the user has already selected the lowest quality, then we will select the quality that’s closest to that.

- Download videos according to the selected video option


- Tap on video download quality selection on the settings page

- Quality option selected

- Tap on the quality settings on videos page

# Screenshots

<img src="https://user-images.githubusercontent.com/960241/130201950-59fdf6a9-7018-46bd-aca8-9a6f9abd414a.png" width="320" height="640">

<img src="https://user-images.githubusercontent.com/960241/130201962-cd955e3d-4106-4433-9d3c-bb01a4573e9c.png" width="320" height="640">
